src/server/start_service.py

```diff
@@ -29,37 +34,131 @@ def create_periodic_introducer_poll_task(
     server,
```

Comment:
Could you refactor this function to its own file? It doesn't really belong in `start_service` (and I'm the one who put it there, so sorry. Never got around to moving it).

Reply:
Agreed